### DISCUSSION TITLE
Delete Button appears on after "Select All" checkbox is selected, even in Resources where the Policy does not allow to delete

### DISCUSSION THREAD

**Discussion by rzuloaga** [laravel/nova-issues#6051] [Ideas & Feature Requests]
- Laravel Version: 4.29.5
- Nova Version: 10.31.0
- PHP Version: 8.2.12
- Database Driver & Version: MySQL 5.7
- Operating System and Version: MacOS Ventura 13.2.1
- Browser type and version: Google Chrome 119
- Reproduction Repository:

### Description:

I have many resources that Laravel policies don't allow me to delete (i.e. `delete()` in that Resource's Model policy is set to return false), and under the previous version of Nova we were using (v.3) it did not allow us to delete that resource.

At the same time, nowadays the resource is not allowed to be deleted in an individual basis: if we mark the checkbox of only one resource on the list, the delete button does not appear:

<img width="133" alt="image" src="https://github.com/laravel/nova-issues/assets/34514565/5ce4814c-75e3-4afb-8da3-66f8e085ceb4">
<img width="240" alt="image" src="https://github.com/laravel/nova-issues/assets/34514565/9284ac0a-7037-44c2-a4a5-9c4b0167f8e4">

But now, if I press the checkbox of "Select All", suddenly the delete button appears in the top of my list. And it shouldn't, because the resource is not allowed to be deleted.

<img width="266" alt="image" src="https://github.com/laravel/nova-issues/assets/34514565/9796be3f-7c1e-4a4a-8bb0-9dc96ad564f5">
<img width="911" alt="image" src="https://github.com/laravel/nova-issues/assets/34514565/e61e6fe4-330e-4886-8e91-cc2e4c32e979">

It's mostly a visual bug, because if I press the button nothing happens. Well, a DELETE REST request is sent, but no deletion actually occurs in the database, as Laravel policies serve as a stopguard.

Tracking Nova source code, I see this in `nova\resources\js\views\Index.vue(589)`:

<img width="605" alt="image" src="https://github.com/laravel/nova-issues/assets/34514565/c692a5ae-3154-41bc-8937-b58dc0953e28">

Looks like it allows the delete button to appear if the checkbox called as "Select All" is marked.

### Detailed steps to reproduce the issue on a fresh Nova installation:

- Create a resource
- Set its Laravel policy to NOT allow deletion
- Go to the Resource Index view and mark the "Select All" on the checkbox dropdown on top of the list.

The delete button (a trash bin) should appear on the top right corner of the list.

**Comment by crynobone** [comment]
You can only request to delete all, however only right before the deletion process we verify if the user has authorization once the model has been hydrated. If we do this during the selection on thousands if not millions of records it would require huge resources on your server each time a user does select all.

**Reply by rzuloaga** [reply]
Could we add an `authorizedToDeleteAll` option to the "Resource", like what we have with `authorizeToDelete` et al.? And it would hide the button even if the "Select All" option is selected.

Having the "Delete" button appear when checking the "Select All" checkbox is causing a lot of confusion to our business people. Even if we insist that "don't worry, the resources will not be deleted".

**Reply by eduardoximenes** [reply]
This problem exists since nova came out, and they don't even care to solve it. It's unbelievable.
